mod_pbxproj/mod_pbxproj.py

```diff
@@ -1282,54 +1320,77 @@ def save_new_format(self, file_name=None, sort=False):
-                    uuids[objs.get(key).get('buildConfigurationList')] = 'Build configuration list for PBXProject "Unity-iPhone"'
-                elif objs.get(key).get('isa')[0:3] == 'PBX':
-                    uuids[key] = objs.get(key).get('isa')[3:-10]
+            obj = objs.get(key)
+            isa = obj.get('isa')
+            if isa in sections:
+                l = sections.get(isa)
+
+            if isa.endswith('BuildPhase'):
+                build_phases[key] = obj
+
+            l.append(tuple([key, obj]))
+            sections[isa] = l
+
+            if 'name' in obj:
+                uuids[key] = obj.get('name')
+            elif 'path' in obj:
+                uuids[key] = obj.get('path')
+            elif isa != 'PBXGroup':
+                if isa == 'PBXProject':
+                    uuids[obj.get('buildConfigurationList')] = 'Build configuration list for %s "%s"'
+                elif isa.startswith('PBX') and isa in self.build_phase_types:
+                    uuids[key] = self.build_phase_types[isa]
+                elif isa.startswith('PBX') and isa != 'PBXGroup':
+                    uuids[key] = isa
                 else:
-                    uuids[key] = 'Build configuration list for PBXNativeTarget "TARGET_NAME"'
+                    uuids[key] = 'Build configuration list for %s "%s"'
 
         ro = self.data.get('rootObject')
         uuids[ro] = 'Project object'
 
         for key in objs:
+            obj = objs.get(key)
+
             # transitive references (used in the BuildFile section)
-            if 'fileRef' in objs.get(key) and objs.get(key).get('fileRef') in uuids:
-                uuids[key] = uuids[objs.get(key).get('fileRef')]
+            if 'fileRef' in obj and obj.get('fileRef') in uuids:
+                uuids[key] = uuids[obj.get('fileRef')]
 
             # transitive reference to the target name (used in the Native target section)
-            if objs.get(key).get('isa') == 'PBXNativeTarget':
-                uuids[objs.get(key).get('buildConfigurationList')] = uuids[objs.get(key).get('buildConfigurationList')].replace('TARGET_NAME', uuids[key])
+            if obj.get('isa') in ['PBXNativeTarget', 'PBXProject']:
+                name = uuids[key]
+                if name == 'Project object':
+                    name = os.path.splitext(os.path.basename(os.path.dirname(self.pbxproj_path)))[0]  # need the product name here
+                uuids[obj.get('buildConfigurationList')] = uuids[obj.get('buildConfigurationList')] % (obj.get('isa'), name)
+
+        build_phase_files = dict()
+        for key, build_phase in _items(build_phases):
+            for file_ref in build_phase.get('files'):
+                build_phase_files[file_ref] = build_phase
 
         self.uuids = uuids
         self.sections = sections
+        self.build_phases = build_phase_files
 
-        out = open(file_name, 'w')
+        if sys.version_info[0] < 3:
+            out = open(file_name, 'w')
```

Comment:
couldn't this be added on the adapter methods above?
_open(file, mode) -> open(file, mode) in 2
-> open(file, mode, encoding) in 3